drivers/serial/uart_bcm2711.c

```diff
@@ -0,0 +1,329 @@
+	uart_write32(dev, BCM2711_MU_IO, ch);
+}
+
+static void bcm2711_mu_lowlevel_init(const struct device *dev, bool skip_baudrate_config,
+			      uint32_t baudrate, uint32_t input_clock)
+{
+	uint32_t divider;
+
+	/* Wait until there is data in the FIFO */
+	while (!bcm2711_mu_lowlevel_can_putc(dev))
+		;
+
+	/* Disable port */
+	uart_write32(dev, BCM2711_MU_CNTL, 0x0);
+
+	/* Disable interrupts */
+	uart_write32(dev, BCM2711_MU_IER, 0x0);
+
+	/* Setup 8bit data width and baudrate */
+	uart_write32(dev, BCM2711_MU_LCR, BCM2711_MU_LCR_8BIT);
+	if (!skip_baudrate_config) {
+		divider = (input_clock / (baudrate * 8));
+		uart_write32(dev, BCM2711_MU_BAUD, divider - 1);
+	}
+
+	/* Enable RX & TX port */
+	uart_write32(dev, BCM2711_MU_CNTL, BCM2711_MU_CNTL_RX_ENABLE | BCM2711_MU_CNTL_TX_ENABLE);
+}
+
+/**
+ * @brief Initialize UART channel
+ *
+ * This routine is called to reset the chip in a quiescent state.
+ * It is assumed that this function is called only once per UART.
+ *
+ * @param dev UART device struct
+ *
+ * @return 0
+ */
+static int uart_bcm2711_init(const struct device *dev)
+{
+	const struct bcm2711_uart_config *uart_cfg = (dev)->config;
+
+	DEVICE_MMIO_MAP(dev, K_MEM_CACHE_NONE);
+	bcm2711_mu_lowlevel_init(dev, 1, uart_cfg->baud_rate, uart_cfg->clocks);
+	return 0;
+}
+
+static void uart_bcm2711_poll_out(const struct device *dev, unsigned char c)
+{
+	bcm2711_mu_lowlevel_putc(dev, c);
+}
+
+static int uart_bcm2711_poll_in(const struct device *dev, unsigned char *c)
+{
+	*c = bcm2711_mu_lowlevel_getc(dev);
+	return 0;
+}
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+
+static int uart_bcm2711_fifo_fill(const struct device *dev,
+				  const uint8_t *tx_data,
+				  int size)
+{
+	unsigned int num_tx = 0U;
+
+	while ((size - num_tx) > 0) {
+		/* Send a character */
+		bcm2711_mu_lowlevel_putc(dev, tx_data[num_tx]);
+		num_tx++;
+	}
+
+	return (int)num_tx;
+}
+
+static int uart_bcm2711_fifo_read(const struct device *dev, uint8_t *rx_data,
+				  const int size)
+{
+	unsigned int num_rx = 0U;
+
+	while ((size - num_rx) > 0) {
+		/* Receive a character */
+		rx_data[num_rx++] = bcm2711_mu_lowlevel_getc(dev);
+	}
+	return num_rx;
+}
+
+static void uart_bcm2711_irq_tx_enable(const struct device *dev)
+{
+	uart_write32(dev, BCM2711_MU_CNTL, BCM2711_MU_CNTL_TX_ENABLE);
+}
+
+static void uart_bcm2711_irq_tx_disable(const struct device *dev)
+{
+	uart_write32(dev, BCM2711_MU_CNTL, (uint32_t)(~BCM2711_MU_CNTL_TX_ENABLE));
+}
+
+static int uart_bcm2711_irq_tx_ready(const struct device *dev)
+{
+	return bcm2711_mu_lowlevel_can_putc(dev);
+}
+
+static void uart_bcm2711_irq_rx_enable(const struct device *dev)
+{
+	uart_write32(dev, BCM2711_MU_CNTL, BCM2711_MU_CNTL_RX_ENABLE);
+}
+
+static void uart_bcm2711_irq_rx_disable(const struct device *dev)
+{
+	uart_write32(dev, BCM2711_MU_CNTL, (uint32_t)(~BCM2711_MU_CNTL_RX_ENABLE));
+}
+
+static int uart_bcm2711_irq_rx_ready(const struct device *dev)
+{
+	return bcm2711_mu_lowlevel_can_getc(dev);
+}
```

Comment:
why are you introducing an entirely new function just to call another function with exactly the same parameter?